src/main/java/BMO.java

```diff
@@ -0,0 +1,321 @@
+
+        String toDoFormat = "^todo\\s+(\\S+(\\s+\\w+)*)$";
+        Pattern toDoPattern = Pattern.compile(toDoFormat);
+        Matcher toDoMatcher = toDoPattern.matcher(input);
+
+        Task newTask;
+        if (deadlineMatcher.matches()) {
+            String task = deadlineMatcher.group(1);
+            String by = deadlineMatcher.group(3);
+
+            LocalDateTime byDateTime = formatDateTime(by);
+
+            newTask = new Deadlines(task, byDateTime);
+        } else if (eventMatcher.matches()) {
+            String task = eventMatcher.group(1);
+            String start = eventMatcher.group(3);
+            String end = eventMatcher.group(5);
+
+            LocalDateTime startDateTime = formatDateTime(start);
+            LocalDateTime endDateTime = formatDateTime(end);
+
+            newTask = new Events(task, startDateTime, endDateTime);
+        } else if (toDoMatcher.matches()) {
+            String task = toDoMatcher.group(1);
+            newTask = new ToDos(task);
+        } else {
+            if (input.startsWith("deadline")) {
+                System.out.println(Constants.errorPrint.deadline());
+            } else if (input.startsWith("todo")) {
+                System.out.println(Constants.errorPrint.todo());
+            } else if (input.startsWith("event")) {
+                System.out.println(Constants.errorPrint.event());
+            } else if (input.isBlank()) {
+                System.out.println(Constants.errorPrint.erroneousAdd());
+            } else {
+                System.out.println(Constants.errorPrint.general());
+            }
+            return;
+        }
+
+        taskLog.add(newTask);
+        String addPrint = "-----------------------------------------\n"
+                + "    added to log: " + newTask + "\n"
+                + "    key in 'log' to view your current task log ^.^\n"
+                + "-----------------------------------------\n";
+        System.out.println(addPrint);
+        return;
+    }
+
+    static void viewLog() {
+        StringBuilder logPrint = new StringBuilder();
+        if (taskLog.isEmpty()) {
+            System.out.println(Constants.emptyLogPrint);
+            return;
+        }
+        for (int i = 0; i < taskLog.size(); i++) {
+            Task currTask = taskLog.get(i);
+            logPrint.append(i + 1).append(". ").append(currTask.getStatusIcon())
+                    .append(" ").append(currTask.toString())
+                    .append("\n");
+        }
+        System.out.println(logPrint.toString());
+        return;
+    }
+
+    static Boolean isIndexCorrect(String input) {
+        if (input.isBlank() || !input.matches("\\d+")) {
+            System.out.println(Constants.errorPrint.noInt());
+            return false;
+        }
+
+        int index = Integer.parseInt((input));
+
+        if (index > taskLog.size() || index <= 0) {
+            System.out.println(Constants.errorPrint.outOfRange());
```

Comment:
Consider changing the naming convention of the printing functions like outOfRange, perhaps use a verb?